src/NUnitEngine/nunit.engine/Services/ExtensionService.cs

```diff
@@ -397,6 +391,11 @@ private void ProcessCandidateAssembly(string filePath, bool fromWildCard)
                     if (!fromWildCard)
                         throw new NUnitEngineException(String.Format("Specified extension {0} could not be read", filePath), e);
                 }
+                catch (NUnitEngineException)
+                {
+                    if (!fromWildCard)
+                        throw;
+                }
```

Comment:
Additional check as we now attempt to read target frameworks in this section, which can throw.